src/data/sources/epicgames/EpicGamesGame.vala

```diff
@@ -0,0 +1,172 @@
+				debug ("New installed game: \tname = %s\t", name);
+			} else {
+
+				debug ("New not installed game: \tname = %s\t", name);
+			}
+			
+			if(state == Game.State.INSTALLED)
+			{
+				remove_tag(Tables.Tags.BUILTIN_UNINSTALLED);
+				add_tag(Tables.Tags.BUILTIN_INSTALLED);
+			}
+			else
+			{
+				add_tag(Tables.Tags.BUILTIN_UNINSTALLED);
+				remove_tag(Tables.Tags.BUILTIN_INSTALLED);
+			}
+			status = new Game.Status(state, this);
+		}
+
+		public EpicGamesGame.from_db(EpicGames src, Sqlite.Statement s)
+		{
+			source = src;
+			id = Tables.Games.ID.get(s);
+			name = Tables.Games.NAME.get(s);
+			info = Tables.Games.INFO.get(s);
+			info_detailed = Tables.Games.INFO_DETAILED.get(s);
+			icon = Tables.Games.ICON.get(s);
+			image = src.legendary_wrapper.get_image(id);//Tables.Games.IMAGE.get(s);
+			install_dir = Tables.Games.INSTALL_PATH.get(s) != null ? FSUtils.file(Tables.Games.INSTALL_PATH.get(s)) : null;
+			executable_path = Tables.Games.EXECUTABLE.get(s);
+			work_dir_path = Tables.Games.WORK_DIR.get(s);
+			compat_tool = Tables.Games.COMPAT_TOOL.get(s);
+			compat_tool_settings = Tables.Games.COMPAT_TOOL_SETTINGS.get(s);
+			arguments = Tables.Games.ARGUMENTS.get(s);
+			last_launch = Tables.Games.LAST_LAUNCH.get_int64(s);
+			playtime_source = Tables.Games.PLAYTIME_SOURCE.get_int64(s);
+			playtime_tracked = Tables.Games.PLAYTIME_TRACKED.get_int64(s);
+			image_vertical = Tables.Games.IMAGE_VERTICAL.get(s);
+
+			platforms.clear();
+			var pls = Tables.Games.PLATFORMS.get(s).split(",");
+			foreach(var pl in pls)
+			{
+				foreach(var p in Platform.PLATFORMS)
+				{
+					if(pl == p.id())
+					{
+						platforms.add(p);
+						break;
+					}
+				}
+			}
+			installers.add(new EpicGamesGame.EpicGamesInstaller(this, id));
+			update_status();
+		}
+
+		public override async void install(Runnable.Installer.InstallMode install_mode=Runnable.Installer.InstallMode.INTERACTIVE)
+		{
+			new GameHub.UI.Dialogs.InstallDialog(this, installers, install_mode, install.callback);
+			yield;
+			update_status();
+		}
+		public override async void uninstall()
+		{
+			((EpicGames)source).legendary_wrapper.uninstall(id);
+			update_status();
+		}
+
+		public override async void run()
+		{
+			((EpicGames)source).legendary_wrapper.run(id);
+		
+		}
+
+		public class EpicGamesInstaller: Runnable.Installer
+		{
+			public EpicGamesGame game;
+			public override string name { owned get { return "TEST"; } }
+	
+			public EpicGamesInstaller(EpicGamesGame game, string id)
+			{
+				this.game = game;
+				id = id;
+				platform = Platform.CURRENT;
+			}
+
+			public override async void install(Runnable runnable, CompatTool? tool=null)
+			{
+				EpicGamesGame? game = null;
+				if(runnable is EpicGamesGame)
+				{
+					game = runnable as EpicGamesGame;
+				}
+
+				EpicGames epic = (EpicGames)(game.source);
+
+				Utils.thread("EpicGamesGame.Installer", () => {
+					game.status = new Game.Status(Game.State.DOWNLOADING, game, null);
```

Comment:
I should add progress bar change here. Should I create own Game.State or use existing in Downloader.